transcom/mymove
# MAIN B-21414 - TOO update for international crate approval requests

## [B-21414](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21414)

INT PR: https://github.com/transcom/mymove/pull/14274

Upstream story: https://github.com/transcom/mymove/pull/14220

## Summary

This adds missing details to International Crating/Uncrating service items that has been requested by the Prime in TOO Move task order page and makes sure it can be approved/denied by the TOO.

- Added to show 'Reason', 'Market', and 'External crate' in the international cratings service item detail.
- Added to show 'Reason' and 'Market' in the international uncratings service item detail.
- Added ` - Standalone` to the International Crating title if `standaloneCrate` is `true`.
- Added logic to `model_to_payload` to show correct `market` depending on the `address.isOconus`

### How to test

1. Follow the steps in this [PR](https://github.com/transcom/mymove/pull/14223) to create international crating/uncrating service items.
2. As a TOO -> go to the move -> move task order tab -> scroll down and you should see them under requested service items table.
3. Verify that these details are showing for both service items.

International Crating:
```
Description: 
Item size: 
Crate size: 
External crate (IT WILL SHOW ONLY IF YOU CHECKED TRUE): 
Market: 
Reason: 
```
International Uncrating:
```
Description: 
Item size: 
Crate size: 
Market: 
Reason: 
```

4. If you selected `Standalone` then it will show in the title of the service item.
5. Click "Approve" or "Reject" to test both scenarios.
6. Verify it works.

## Screenshots
![image](https://github.com/user-attachments/assets/af1b7156-a9eb-41f3-b74e-8f624d7b7382)
![image](https://github.com/user-attachments/assets/f5f1a978-adc5-474e-8090-ecabf98823ac)
![image](https://github.com/user-attachments/assets/216c45eb-60be-4dcd-b9f2-b0759b678a8e)
![image](https://github.com/user-attachments/assets/7a8682fa-3c94-4f74-97b1-a90c9a2bec2e)
